tests/runtimes/test_application.py

```diff
@@ -255,6 +258,17 @@ def test_deploy_reverse_proxy_image(rundb_mock, igz_version_mock):
     assert mlrun.runtimes.ApplicationRuntime.reverse_proxy_image
 
 
+def test_application_from_local_file_validation():
+    project = mlrun.get_or_create_project("test-application")
+    func_path = assets_path / "sample_function.py"
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as exc:
+        project.set_function(func=str(func_path), name="my-app", kind="application")
+    assert str(exc.value) == (
+        "Embedding a code file is not supported for application runtime. "
+        "Code files should be specified via project/function source."
+    )
```

Comment:
```suggestion
    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError, match="Embedding a code file ..."):
        project.set_function(func=str(func_path), name="my-app", kind="application")
```

`pytest.raises` already supports matching exception text representation. Not critical, FYI.